bin/p2-rm/rm.go

```diff
@@ -0,0 +1,174 @@
+	return rm
+}
+
+func (rm *P2RM) configureStorage(client consulutil.ConsulClient, labeler labels.ApplicatorWithoutWatches) {
+	rm.Client = client
+	rm.Store = kp.NewConsulStore(client)
+	rm.RCStore = rcstore.NewConsul(client, labeler, 5)
+	rm.Labeler = labeler
+	rm.PodStore = podstore.NewConsul(client.KV())
+}
+
+func (rm *P2RM) checkForManagingReplicationController() (bool, fields.ID, error) {
+	podLabels, err := rm.Labeler.GetLabels(labels.POD, rm.LabelID)
+	if err != nil {
+		return false, "", fmt.Errorf("unable to check node for labels: %v", err)
+	}
+
+	if podLabels.Labels.Has(rc.RCIDLabel) {
+		return true, fields.ID(podLabels.Labels.Get(rc.RCIDLabel)), nil
+	}
+
+	return false, "", nil
+}
+
+func (rm *P2RM) decrementDesiredCount(id fields.ID) error {
+	session, _, err := rm.Store.NewSession(sessionName(id), nil)
+	if err != nil {
+		return fmt.Errorf("Unable to get consul session: %v", err)
+	}
+
+	rcLock, err := rm.RCStore.LockForMutation(id, session)
+	if err != nil {
+		return fmt.Errorf("Unable to lock RC for mutation: %v", err)
+	}
+	defer rcLock.Unlock()
+
+	err = rm.RCStore.Disable(id)
+	if err != nil {
+		return fmt.Errorf("Could not disable RC %s: %v", id, err)
+	}
+
+	err = rm.deletePod()
+	if err != nil {
+		return fmt.Errorf("Unable to delete pod. Please re-run this command to clean up: %v", err)
+	}
+
+	err = rm.RCStore.AddDesiredReplicas(id, -1)
+	if err != nil {
+		return fmt.Errorf("Unable to decrement RC count: %v", err)
+	}
+
+	err = rm.RCStore.Enable(id)
+	if err != nil {
+		return fmt.Errorf("Could not enable RC %s: %v", id, err)
+	}
+
+	if err = session.Destroy(); err != nil {
+		return fmt.Errorf("Unable to destroy consul session: %v", err)
+	}
+
+	return nil
+}
+
+func (rm *P2RM) deletePod() error {
+	if rm.PodUniqueKey == "" {
+		err := rm.deleteLegacyPod()
+		if err != nil {
+			return err
+		}
+	} else {
+		err := rm.deleteUUIDPod(rm.PodUniqueKey, rm.PodName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return rm.removePodLabels()
+}
+
+func (rm *P2RM) deleteLegacyPod() error {
+	_, err := rm.Store.DeletePod(kp.INTENT_TREE, rm.NodeName, types.PodID(rm.PodName))
+	if err != nil {
+		return fmt.Errorf("unable to remove pod: %v", err)
+	}
+
+	return nil
+}
+
+func (rm *P2RM) deleteUUIDPod(podUniqueKey types.PodUniqueKey, podID types.PodID) error {
+	// Sanity check that the passed pod ID matches the pod being deleted
+	pod, err := rm.PodStore.ReadPod(podUniqueKey)
+	if err != nil {
+		return fmt.Errorf("Could not verify that pod %s matches pod ID of %s: %s", podUniqueKey, podID, err)
+	}
+
+	if pod.Manifest.ID() != podID {
+		return fmt.Errorf("pod %s has a podID of %s, but %s was passed as the --pod option", podUniqueKey, pod.Manifest.ID(), podID)
+	}
+
+	err = rm.PodStore.Unschedule(podUniqueKey)
+	if err != nil {
+		return fmt.Errorf("Unable to unschedule pod: %s", err)
+	}
+	return nil
+}
+
+func (rm *P2RM) removePodLabels() error {
+	err := rm.Labeler.RemoveAllLabels(labels.POD, rm.LabelID)
```

Comment:
answering own question: yes. `rm.LabelID` gets set to `podUniqueKey.String()` for a UUID pod.